### PR TITLE
Clear 'opt-out' flags for 'newsletter' opt-ins

### PR DESCRIPTION
**Pull request by awg-github**

### CODE REVIEW

File: api/v3/Engage/Signpetition.php [hunk 1]
```diff
@@ -109,6 +109,9 @@ function _civicrm_api3_engage_signpetition_process($params) {
       CRM_Gpapi_Processor::addToGroup($contact_id, 'Community NL');
     }
 
+    //remove "Opt Out" and "do not email"
+    CRM_Gpapi_Processor::enableSubscription($contact_id);
```
Comment (pfigel) [inline api/v3/Engage/Signpetition.php]: This should happen only in the `if` above.

File: api/v3/Newsletter/Subscribe.php [hunk 2]
```diff
@@ -63,40 +63,22 @@ function _civicrm_api3_newsletter_subscribe_process($params) {
...
-      ]);
-      civicrm_api3('GroupContact', 'create', [
-        'check_permissions' => 0,
-        'contact_id'        => $contact_id,
-        'group_id'          => $volunteer_group['id']
-      ]);
+      CRM_Gpapi_Processor::addToGroup($contact_id, 'Ehrenamtliche NL');
+
       $subscribed = TRUE;
     }
```
Comment (pfigel) [inline api/v3/Newsletter/Subscribe.php]: `Newsletter.subscribe` does not use the new `CRM_Gpapi_Processor::enableSubscription` method, should be DRY'd up too.

File: api/v3/Engage/Signpetition.php [hunk 3]
```diff
@@ -109,6 +109,9 @@ function _civicrm_api3_engage_signpetition_process($params) {
       CRM_Gpapi_Processor::addToGroup($contact_id, 'Community NL');
     }
 
+    //remove "Opt Out" and "do not email"
```
Comment (awg-github) [inline api/v3/Engage/Signpetition.php]: Above you can see comment (GP-463: "der Group "Donation Info" Eintrag soll immer gesetzt werden..."). If contact is always added to group, that's why we always call 'enableSubscription' method.

Comment (pfigel) [reply api/v3/Engage/Signpetition.php]: Will check this, my best guess is that we process "Donation Info" opt-in under a different legal basis here compared to `Newsletter.subscribe`, hence no explicit permission to clear suppression flags.

File: api/v3/Engage/Signpetition.php (hunk 1 above)
Comment (awg-github) [reply api/v3/Engage/Signpetition.php]: Above you can see comment (GP-463: "der Group "Donation Info" Eintrag soll immer gesetzt werden..."). If contact is always added to group, that's why we always call 'enableSubscription' method.

Comment (awg-github) [reply api/v3/Engage/Signpetition.php]: If you want we can move `enableSubscription` to `if` body.

Comment (pfigel) [reply api/v3/Engage/Signpetition.php]: As discussed in Redmine: Yes, please move it to the `newsletter` if block.

**Comment by pfigel** [comment]
thanks @awg-github!

**Pull request merged**